howto/port/install-as-system-app.md

```diff
@@ -0,0 +1,75 @@
+Normally, a user app is signed by the developer and restricted to fewer permissions at runtime while a system app is usually [signed with the platform key](https://source.android.com/devices/tech/ota/sign_builds) when building an Android image, it's pre-installed under the system partition and running with some ["signature" protection level permissions](https://developer.android.com/guide/topics/manifest/permission-element.html#plevel) in the system process of Android container. An application must be running as a system app if it
+
+- requires to access Android hidden APIs
+- gains some signature protection level permissions
+
+Anbox Cloud allows a user app to be installed as a system app in Android container. This can be done by using the `aam install-system-app` command.
+
+### Make a system app
+To be a system app running in the system process of Android system, you must add the attribute `android:sharedUserId="android.uid.system"` into the `<manifest>` tag in the AndroidManifest.xml file of your Android project.
+
+```xml
+...
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    package="<package_name>"
+    android:sharedUserId="android.uid.system">
+...
+```
+
+Then build and sign the application alongside with other Android applications when building your Android image or sign it with [Android Studio](https://developer.android.com/studio/publish/app-signing). For the latter approach, the system platform key is not required and you can use the keys that are generated from Android Studio to sign the application.
+
+### Create an addon
+
+By creating an addon and invoking the `aam install-system-app` command in a `pre-start` hook, the signed APK can be installed as a system app in the Android container.
+
+Following the [tutorial](https://discourse.ubuntu.com/t/creating-an-addon/25284) and create the basic layout of an addon, which has a manifest.yaml file and hooks folder prepared in advance. Assumed that the APK file is named `app.apk`, place the APK into the addon folder and create a `pre-start` hook with the following content.
+
+```bash
+#!/bin/bash -ex
+
+# Only install the APK as a system app when bootstrapping an application.
+if  [ "$CONTAINER_TYPE" = "regular" ]; then
+  exit 0
+fi
+
+aam install-system-app \
+  --apk="${ADDON_DIR}/app.apk \
```

Comment:
```suggestion
  --apk="${ADDON_DIR}"/app.apk \
```

Reply:
Will fix. 
I think we should show the full addon here rather than referencing another howto. That would make this self contained, ideas?
A: I avoid doing that. The reason is that if we do that, we may replicate some other bits which existed in the tutorial section. For this page, I just want to highlight the stuff that people need to do to install apk as a system app without introducing some common knowledge/syntaxes that can be found from the tutorial which would be the first place people could get hands on the addon creation.

Reply:
Ok, fine for me

Reply:
I agree with Gary - we shouldn't duplicate content. And advanced users will know how to create an addon without further instructions anyway.